src/ClusterProviders/Proto.Cluster.SingleRemoteInstance/SingleRemoteInstanceProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Proto.Remote;
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Proto.Cluster.SingleRemoteInstance
+{
+    public class SingleRemoteInstanceProvider : IClusterProvider
+    {
+        const string KINDS_RESPONDER = "remote_kinds_responder";
+        readonly TimeSpan _timeout = TimeSpan.FromSeconds(10);
+
+        private readonly string _serverHost;
+        private readonly int _serverPort;
+        private readonly string _serverAddress;
+
+        private string[] _kinds;
+        private IMemberStatusValue _okStatus;
+        private IMemberStatusValue _koStatus;
+        private bool _isServer;
+        private bool _shutdown;
+
+        public SingleRemoteInstanceProvider(string serverHost, int serverPort)
+        {
+            _serverHost = serverHost;
+            _serverPort = serverPort;
+            _serverAddress = $"{serverHost}:{serverPort}";
+
+            var props = Props.FromFunc(ctx =>
+            {
+                if ((ctx.Message is GetKinds) && ctx.Sender != null)
+                    ctx.Respond(new GetKindsResponse { Kinds = { _kinds } });
+
+                return Actor.Done;
+            });
+
+            Serialization.RegisterFileDescriptor(ProtosReflection.Descriptor);
+            //RootContext.Empty.SpawnNamed(props, KINDS_RESPONDER);
+
+            Remote.Remote.RegisterKnownKind(KINDS_RESPONDER, props);
+        }
+
+        public Task DeregisterMemberAsync() => Actor.Done;
+
+        public Task RegisterMemberAsync(string clusterName, string host, int port, string[] kinds, IMemberStatusValue statusValue, IMemberStatusValueSerializer serializer)
+        {
+            _kinds = kinds;
+            _okStatus = serializer.FromValueBytes(Encoding.UTF8.GetBytes("Ok!"));
+            _koStatus = serializer.FromValueBytes(Encoding.UTF8.GetBytes("Ko!"));
+
+            _isServer = host == _serverHost && port == _serverPort;
+
+            return Actor.Done;
+        }
+
+        public void MonitorMemberStatusChanges()
+        {
+            var t = new Thread(_ =>
```

Comment:
👍 I was going to say it could be done using Async Await to not hog a thread.
But this is better as if the threadpool is under stress, the notifications will still run at the correct intervals.